### PR TITLE
Added the values.yaml and tried creating an integration testing script.

### PR DESCRIPTION
**Pull request by Senjuti256**
I have added the test/helper/helper.sh file to create the test environment. Also I have added the values.yaml file.

I tried to create a basic structure of the integration testing script with its basic functionalities.
In the integration testing script I tried to ensure that the Skopeo copy task works as expected and that the image is successfully copied from the source registry to the destination registry. It sets up a test environment, deploys the Skopeo copy task, and executes a test case by running the task with the specified input parameters. After the task completes, the test script verifies that the image has been copied to the destination registry by inspecting the destination image using Skopeo. If the destination image matches the expected image, the integration test passes else it fails, indicating that there may be a problem with the Skopeo copy task or the input parameters.

### CODE REVIEW

File: test/e2e/e2e.bats [hunk 1]
```diff
@@ -30,7 +30,7 @@ if ! skopeo inspect --creds "$SOURCE_AUTH_USERNAME:$SOURCE_AUTH_PASSWORD" "docke
 fi
 
 # Run the skopeo-copy task
-if ! tekton pipeline start skopeo-copy -p SOURCE_REGISTRY="$SOURCE_REGISTRY" \
+if ! tkn pipeline start skopeo-copy -p SOURCE_REGISTRY="$SOURCE_REGISTRY" \
```
Comment (otaviof) [inline test/e2e/e2e.bats:33]: Please use a construction like this:
```bash
run false
assert_success
```
For instance: https://github.com/openshift-pipelines/task-git/blob/fbd3c8b15b10b0deda87538a8cbfb53aada6b1d3/test/e2e/e2e.bats#L11-L19

**Review by otaviof — CHANGES_REQUESTED** [review]
Please consider the bats example.

**Comment by anandxkumar** [comment]
@Senjuti256 inorder to resolve the branch conflict, pull the main branch (remote repo) to your current repo, so that the `task-sc.yaml` is the latest and your instance is not conflicting with it.

File: templates/task-sc.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
```
Comment (otaviof) [inline templates/task-sc.yaml:4]: Make sure the [formatting is correct](https://redhat-internal.slack.com/archives/C04J0SVUDTL/p1679555852819269), please.

File: templates/task-sc.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            description: |
+                Source Registry Password for authentication
+        -   name: DESTINATION_AUTH_USERNAME
+            type: string
+            description: |
+                Destination Registry Username for authentication
+        -   name: DESTINATION_AUTH_PASSWORD
+            type: string
+            description: |
+                Destination Registry Password for authentication
```
Comment (otaviof) [inline templates/task-sc.yaml:36]: This type of parameters should be added on the next step, when we start using optional workspaces for advanced authentication cases. Let's focus first on using what Tekton Pipeline provides out of the box.

File: templates/task-sc.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            skopeo login docker.io -u=$(params.SOURCE_AUTH_USERNAME) -p=$(params.SOURCE_AUTH_PASSWORD)
+    -   name: login-dest
+        image: quay.io/skopeo/stable:latest
+        script: |
+            skopeo login docker.io -u=$(params.DESTINATION_AUTH_USERNAME) -p=$(params.DESTINATION_AUTH_PASSWORD)
+    -   name: copy
+        image: quay.io/skopeo/stable:latest
+        script: |
+            skopeo copy --src-creds $(params.SOURCE_AUTH_USERNAME):$(params.SOURCE_AUTH_PASSWORD) \
+            docker://$(params.SOURCE_REGISTRY) --dest-creds $(params.DESTINATION_AUTH_USERNAME):$(params.DESTINATION_AUTH_PASSWORD) \
```
Comment (otaviof) [inline templates/task-sc.yaml:50]: What does `docker://` protocol means for the container image location?

File: test/e2e/e2e.bats [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+# Define test case
+read -p "Enter source registry URL: " SOURCE_REGISTRY
+read -p "Enter destination registry URL: " DESTINATION_REGISTRY
+read -p "Enter source registry username: " SOURCE_AUTH_USERNAME
+read -p "Enter source registry password: " -s SOURCE_AUTH_PASSWORD
+echo  "Source registry Credentials are taken as input"
+
+read -p "Enter destination registry username: " DESTINATION_AUTH_USERNAME
+read -p "Enter destination registry password: " -s DESTINATION_AUTH_PASSWORD
+echo  "Destination registry Credentials are taken as input"
```
Comment (otaviof) [inline test/e2e/e2e.bats:22]: The configuration needed for testing should come from the environment variables, for instance:
- https://github.com/otaviof/tkn-buildpacks-task/blob/02293f1720a8f59a3051dad98e2e8b99d27fe889/Makefile#L10-L14
- https://github.com/otaviof/tkn-buildpacks-task/blob/02293f1720a8f59a3051dad98e2e8b99d27fe889/test/e2e/e2e.bats#L40-L42

Then running the tests with alternative settings becomes:
```bash
$ make test-e2e VARIABLE="new-value"
```

Where `${VARIABLE}` will end up on the scripts we need.

File: values.yaml [hunk 6]
```diff
@@ -1 +1,24 @@
...
+  securityContext:
+    runAsNonRoot: true
+    runAsUser: 65532
+  resources:
+    limits:
+      cpu: 100m
+      memory: 256Mi
+    requests:
+      cpu: 100m
+      memory: 256Mi
```
Comment (otaviof) [inline values.yaml:22]: This needs to reflect the current project.

**Review by otaviof — CHANGES_REQUESTED** [review]
You've made very good progress, thanks! Let's start to address more details of how the project and tests are structured.

File: templates/task-sc.yaml (hunk 3 above)
Comment (anandxkumar) [reply templates/task-sc.yaml:36]: Right, I'm looking into the authentication part!

File: values.yaml (hunk 6 above)
Comment (Senjuti256) [reply values.yaml:22]: Ok I will update it.

File: test/e2e/e2e.bats [hunk 7]
```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bats
+
+
+# Set up test environment
+
+source ./test/helper/helper.sh
+
+
+# Deploy task to test environment
+kubectl apply -f templates/task-sc.yaml
```
[the remark this answers is not in the document]
Comment (Senjuti256) [reply test/e2e/e2e.bats:10]: Ok I am trying to do it.

File: templates/task-sc.yaml (hunk 4 above)
Comment (Senjuti256) [reply templates/task-sc.yaml:50]: I am not sure this task-sc.yaml file is not written by me. It was merged few days ago.

Comment (anandxkumar) [reply templates/task-sc.yaml:50]: It is basically used for registry implementing the "Docker Registry HTTP API V2". So like if we want to work with registries like docker,  quay, federa, etc. All examples I looked upto were using this protocol for specifying the image.

**Comment by otaviof** [comment]
I'm closing this in favor of #6 🙏

**Pull request closed without merging**